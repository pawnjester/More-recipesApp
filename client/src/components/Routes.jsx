import React from 'react';
import { Switch, Route, Router } from 'react-router-dom';
import history from '../utils/history';
import HomeComponent from './Homepage';
import LoginPageComponent from './login/LoginPage';
import RecipePageComponent from './recipe/addRecipe/RecipePage';
import DetailComponent from './recipe/Detail';
import NotFound from './NotFound';
import ProfileComponent from './user/profile';
import Favorite from './recipe/FavoriteRecipe';
import SearchPageComponent from './SearchPage';
import PasswordResetComponent from './PasswordReset';
import PasswordChangeFormComponent from './PasswordChangeForm';
import RedirectToLoginComponent from './RedirectToLogin';
import ChangePasswordForm from './ChangePasswordForm';
import AllRecipesComponent from './recipe/AllRecipes';

import requireAuth from '../utils/requireAuth';


const Routes = () => (
  <Router history={history}>

    <Switch>
      <Route exact path="/" component={HomeComponent} />
      <Route path="/signin" component={LoginPageComponent} />
      <Route path="/recipes" component={requireAuth(RecipePageComponent)} />
      <Route path="/detail/:recipeId" component={requireAuth(DetailComponent)} />
      <Route path="/favorites" component={requireAuth(Favorite)} />
      <Route path="/profile" component={requireAuth(ProfileComponent)} />
      <Route path="/search" component={SearchPageComponent} />
      <Route path="/all-recipes" component={requireAuth(AllRecipesComponent)} />
      <Route path="/reset_password/" component={PasswordResetComponent} />
      <Route
        path="/auth/reset_password/:token"
        component={PasswordChangeFormComponent}
      />
      <Route path="/newLogin" component={RedirectToLoginComponent} />
      <Route
        path="/change-password"
        component={requireAuth(ChangePasswordForm)}
      />
      <Route path="*" component={NotFound} />
    </Switch>
  </Router>

);


export default Routes;
