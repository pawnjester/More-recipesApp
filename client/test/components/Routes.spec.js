import React from 'React';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import expect from 'expect';
import Routes from '../../src//components/Routes';


configure({ adapter: new Adapter() });

describe('Test For Routes', () => {
  it('should test for routes', () => {
    const wrapper = shallow(<Routes /> );
    expect(wrapper).toMatchSnapshot()
  })

})
