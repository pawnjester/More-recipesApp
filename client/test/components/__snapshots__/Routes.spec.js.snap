// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test For Routes should test for routes 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "blank",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/signin"
    />
    <Route
      component={[Function]}
      path="/recipes"
    />
    <Route
      component={[Function]}
      path="/detail/:recipeId"
    />
    <Route
      component={[Function]}
      path="/favorites"
    />
    <Route
      component={[Function]}
      path="/profile"
    />
    <Route
      component={[Function]}
      path="/search"
    />
    <Route
      component={[Function]}
      path="/all-recipes"
    />
    <Route
      component={[Function]}
      path="/reset_password/"
    />
    <Route
      component={[Function]}
      path="/auth/reset_password/:token"
    />
    <Route
      component={[Function]}
      path="/newLogin"
    />
    <Route
      component={[Function]}
      path="/change-password"
    />
    <Route
      component={[Function]}
      path="*"
    />
  </Switch>
</Router>
`;
